### DISCUSSION TITLE
Why the key is "readability-identifier-naming.TypeAliasCase" has the value "camelCase"?

### DISCUSSION THREAD

**Discussion by shrek1402** [carbon-language/carbon-lang#2267] [Getting started]
Why are the keys "readability-identifier-naming.TypeAliasCase" and "readability-identifier-naming.TypedefCase" has the value "camelCase"?

```
 struct Iterator {
    // NOLINTNEXTLINE(readability-identifier-naming)
    using value_type = typename Node::ValueType;
    // NOLINTNEXTLINE(readability-identifier-naming)
    using difference_type = std::ptrdiff_t;
    // NOLINTNEXTLINE(readability-identifier-naming)
    using pointer = const value_type*;
    // NOLINTNEXTLINE(readability-identifier-naming)
    using reference = const value_type&;
    // NOLINTNEXTLINE(readability-identifier-naming)
    using iterator_category = std::forward_iterator_tag;

    ...
};
```

Wouldn't it be logical to use the value "lower_case"? Then you wouldn't have to use "// NOLINTNEXTLINE(readability-identifier-naming)" often.

**Comment by jonmeow** [comment]
These exceptions are specifically for STL API compatibility. In general, we're trying to use https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md#general-naming-rules to try to match what we're considering for Carbon. Switching to lower case would force us to make everything else exceptions, and STL API compatibility makes up a minority of naming decisions.
